base_layer/wallet_ffi/src/lib.rs

```diff
@@ -2566,249 +2647,179 @@ pub unsafe extern "C" fn wallet_destroy(wallet: *mut TariWallet) {
-    use tari_wallet::{testnet_utils::random_string, transaction_service::storage::database::TransactionStatus};
-    use tempdir::TempDir;
+    unsafe extern "C" fn discovery_process_complete_callback(_tx_id: c_ulonglong, _result: bool) {
+        assert!(true);
+    }
 
-    fn type_of<T>(_: T) -> String {
-        std::any::type_name::<T>().to_string()
+    unsafe extern "C" fn received_tx_callback_bob(tx: *mut TariPendingInboundTransaction) {
+        assert_eq!(tx.is_null(), false);
+        assert_eq!(
+            type_of((*tx).clone()),
+            std::any::type_name::<TariPendingInboundTransaction>()
+        );
+        pending_inbound_transaction_destroy(tx);
     }
 
-    unsafe extern "C" fn completed_callback(tx: *mut TariCompletedTransaction) {
+    unsafe extern "C" fn received_tx_reply_callback_bob(tx: *mut TariCompletedTransaction) {
         assert_eq!(tx.is_null(), false);
         assert_eq!(
             type_of((*tx).clone()),
             std::any::type_name::<TariCompletedTransaction>()
         );
-        println!("{:?}", (*tx).status);
-        assert_eq!((*tx).status, TransactionStatus::Broadcast);
+        assert_eq!((*tx).status, TransactionStatus::Completed);
         completed_transaction_destroy(tx);
     }
 
-    unsafe extern "C" fn inbound_callback(tx: *mut TariPendingInboundTransaction) {
+    unsafe extern "C" fn received_tx_finalized_callback_bob(tx: *mut TariCompletedTransaction) {
         assert_eq!(tx.is_null(), false);
         assert_eq!(
             type_of((*tx).clone()),
-            std::any::type_name::<TariPendingInboundTransaction>()
+            std::any::type_name::<TariCompletedTransaction>()
         );
-        pending_inbound_transaction_destroy(tx);
+        assert_eq!((*tx).status, TransactionStatus::Completed);
+        completed_transaction_destroy(tx);
     }
 
-    unsafe extern "C" fn mined_callback(tx: *mut TariCompletedTransaction) {
+    unsafe extern "C" fn broadcast_callback_bob(tx: *mut TariCompletedTransaction) {
         assert_eq!(tx.is_null(), false);
         assert_eq!(
             type_of((*tx).clone()),
             std::any::type_name::<TariCompletedTransaction>()
         );
-        assert_eq!((*tx).status, TransactionStatus::Mined);
+        assert_eq!((*tx).status, TransactionStatus::Broadcast);
         completed_transaction_destroy(tx);
     }
 
-    unsafe extern "C" fn broadcast_callback(tx: *mut TariCompletedTransaction) {
+    unsafe extern "C" fn mined_callback_bob(tx: *mut TariCompletedTransaction) {
         assert_eq!(tx.is_null(), false);
         assert_eq!(
             type_of((*tx).clone()),
             std::any::type_name::<TariCompletedTransaction>()
         );
-        assert_eq!((*tx).status, TransactionStatus::Broadcast);
+        assert_eq!((*tx).status, TransactionStatus::Mined);
         completed_transaction_destroy(tx);
     }
 
+    unsafe extern "C" fn discovery_process_complete_callback_bob(_tx_id: c_ulonglong, _result: bool) {
```

Comment:
Consider a pointer to a transaction struct instead of c_ulonglong for the same reasons as above.

Additionally, it will be in line with all other callbacks.

Reply:
On this one I disagree because the result of the discovery process could fail and then you don't have a transaction struct to return just the tx_id that failed.

Reply:
I do see your point. Agreed